File: examples/typescript/sdk_example.ts
export async function generate(prompt: string, maxTokens = 128, apiBase = '') {
const r = await fetch(`${apiBase}/v1/generate`, { method: 'POST', headers: { 'Content-Type': 'application/json' },
body: JSON.stringify({ prompt, max_tokens: maxTokens }) });
if (!r.ok) throw new Error(`HTTP ${r.status}`); return r.json();
}

File: examples/python/sdk_example.py
"""Minimal Python SDK example."""

import os
import requests  # type: ignore[import-untyped]

API = os.getenv("FACTSYNTH_API", "http://localhost:8000")

def generate(prompt: str, max_tokens: int = 128):
    r = requests.post(f"{API}/v1/generate", json={"prompt": prompt, "max_tokens": max_tokens}, timeout=60)
    r.raise_for_status()
    return r.json()

if __name__ == "__main__":
    print(generate("hello world"))
